code.gs

```diff
@@ -1389,20 +1406,29 @@ function main (forceDate) {
       .concat(isRunningOutdatedVersion() ? ['<br/><br/><b>', htmlEscape(bodySuffixes[2]), ' <a href="', baseGitHubProjectURL, 'releases/latest', '">', htmlEscape(bodySuffixes[3]), '</a>.</b></p>'] : ['</p>'])
       .join('');
 
-    // ...send the email notification.
-    log.add('Sending email...', 'info');
-    if (!settings.debug.log.logInstedOfMail) {
-      sendEmail(subject, body, htmlBody, inlineImages)
-    } else {
+    // ...log (if selected) and return mail content.
+    if (logEmail) {
       log.add('Subject: ' + subject);
       log.add('Content: ' + body);
       //log.add('Content HTML: ' + htmlBody); //is it needed?
     }
+
+    var emailData = {
+      'subject': subject, 
+      'body': body, 
+      'htmlBody': htmlBody, 
+      'inlineImages': inlineImages
+    }
+    
+    return emailData;
   }
-  // Send the log if the debug options say so.
-  log.sendEmail(settings.user.notificationEmail, settings.user.emailSenderName);
+  
+  return null;
```

Comment:
This `return null` is dangerously hidden: I'd make it easier to see by changing [line R1395](https://github.com/GioBonvi/GoogleContactsEventsNotifier/pull/58/commits/ac7c613c1c1ac835e32a37920a77be22aa0fa607#diff-deb2c49f6ef99255c9830fe80e966b46R1395) to:

```javascript
if (bodyBuilder.length === 0) {
  return null;
} else {
  // build and return the email notification.
```